src/intermediate/Instruction.cpp

```diff
@@ -368,22 +368,30 @@ bool IntermediateInstruction::writesLocal(const Local* local) const
 
 void IntermediateInstruction::replaceLocal(const Local* oldLocal, const Local* newLocal, const LocalUse::Type type)
 {
-	if(oldLocal == newLocal)
+	replaceValue(oldLocal->createReference(), newLocal->createReference(), type);
+}
+
+void IntermediateInstruction::replaceLocal(const Local *oldLocal, const Value newValue, LocalUse::Type type) {
+	replaceValue(oldLocal->createReference(), newValue, type);
+}
+
+void IntermediateInstruction::replaceValue(const Value oldValue, const Value newValue, LocalUse::Type type) {
+	if(newValue == oldValue)
 		return;
-	if(has_flag(type, LocalUse::Type::WRITER) && output && output->hasLocal(oldLocal))
+	if(has_flag(type, LocalUse::Type::WRITER) && output && output == oldValue)
 	{
 		removeAsUserFromValue(output.value(), LocalUse::Type::WRITER);
```

Comment:
These 3 lines can be replaced with `setOutput(newValue)` which already removes the usage from the previous output and adds it to the new one.

Reply:
To use `setOutput`, I need to change from range-base for-loop to ordinary for-loop to use index of arguments.
I think current implementation is better than it.